crates/bevy_reflect/src/enums/dynamic_enum.rs

```diff
@@ -54,14 +54,14 @@ impl From<()> for DynamicVariant {
 /// # Example
 ///
 /// ```
-/// # use bevy_reflect::{DynamicEnum, DynamicVariant, Reflect};
+/// # use bevy_reflect::{DynamicEnum, DynamicVariant, Reflect, ReflectTypeName};
 ///
 /// // The original enum value
 /// let mut value: Option<usize> = Some(123);
 ///
 /// // Create a DynamicEnum to represent the new value
 /// let mut dyn_enum = DynamicEnum::new(
-///   Reflect::type_name(&value),
+///   ReflectTypeName::type_name(&value).as_ref(),
```

Comment:
These changes can be undone due to the most recent changes that removed `ReflectTypeName`.